incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -17,24 +21,77 @@ const (
 	HierarchyServingPath = "/validate-hnc-x-k8s-io-v1alpha1-hierarchies"
 )
 
+// Note: the validating webhook FAILS CLOSED. This means that if the webhook goes down, all further
+// changes to the hierarchy are forbidden. However, new objects will still be propagated according
+// to the existing hierarchy (unless the reconciler is down too).
+//
 // +kubebuilder:webhook:path=/validate-hnc-x-k8s-io-v1alpha1-hierarchies,mutating=false,failurePolicy=fail,groups="hnc.x-k8s.io",resources=hierarchies,verbs=create;update,versions=v1alpha1,name=hierarchies.hnc.x-k8s.io
 
 type Hierarchy struct {
 	Log     logr.Logger
+	Forest  *forest.Forest
 	client  client.Client
 	decoder *admission.Decoder
 }
 
 func (v *Hierarchy) Handle(ctx context.Context, req admission.Request) admission.Response {
-	log := v.Log.WithValues("namespace", req.Namespace)
-	log.Info("Validating")
-	hier := &tenancy.Hierarchy{}
-	err := v.decoder.Decode(req, hier)
+	log := v.Log.WithValues("ns", req.Namespace)
+
+	if isHNCServiceAccount(req.AdmissionRequest.UserInfo) {
```

Comment:
Why HNC SA can do anything even when he potentially can create a cycle?

Reply:
Because the object might _already_ be in a bad state - that is, a state that would not pass validation. But the controller would like to update `hier.status` to _show_ that it's in a bad state.

Is there a better way to do this?

Reply:
I'd suggest HNC SA delete/recreate hier CR if it is in bad state.

Reply:
But the bad state could be set by the user, so we can't simply delete and recreate it.

I feel like I've heard that there's a way to update the status without the spec, why don't I open an issue for that.

Reply:
Filed #80